### PR TITLE
Fix create new resource group

### PR DESCRIPTION
**Pull request by oneonestar**
## Description
Create resource group was broken after #287.
`resource_group_id` should be generated by database with AUTO_INCREMENT in DDL.
Tests didn't cover create multiple resource groups so we missed that.

## Release notes
(X) Release notes are required, with the following suggested text:

```markdown
* Fix create new resource group
```

### CODE REVIEW

File: gateway-ha/src/test/java/io/trino/gateway/ha/router/TestResourceGroupsManager.java [hunk 1]
```diff
@@ -50,27 +50,35 @@ public void testCreateResourceGroup()
     {
         ResourceGroupsDetail resourceGroup = new ResourceGroupsDetail();
 
-        resourceGroup.setResourceGroupId(0L);
+        resourceGroup.setResourceGroupId(0L);   // This should be ignored. Id are generated by database (AUTO_INCREMENT) starting from 1.
```
Comment (mosabua) [inline gateway-ha/src/test/java/io/trino/gateway/ha/router/TestResourceGroupsManager.java:53]: What do you mean .. it should be ignored? I think it is ignored right?
```suggestion
        resourceGroup.setResourceGroupId(0L);   // The value is ignored. IDs are generated by the database (AUTO_INCREMENT) starting from 1.
```

File: gateway-ha/src/test/java/io/trino/gateway/ha/router/TestResourceGroupsManager.java [hunk 2]
```diff
@@ -50,27 +50,35 @@ public void testCreateResourceGroup()
...
         resourceGroup.setJmxExport(true);
         resourceGroup.setSoftMemoryLimit("80%");
-
-        ResourceGroupsDetail newResourceGroup = resourceGroupManager.createResourceGroup(resourceGroup,
+        ResourceGroupsDetail adminResourceGroup = resourceGroupManager.createResourceGroup(resourceGroup,
                 null);
+        assertThat(adminResourceGroup).isEqualTo(resourceGroup);
 
-        assertThat(newResourceGroup).isEqualTo(resourceGroup);
+        resourceGroup.setResourceGroupId(0L);   // Id are generated by database (AUTO_INCREMENT). This will be 2.
```
Comment (mosabua) [inline gateway-ha/src/test/java/io/trino/gateway/ha/router/TestResourceGroupsManager.java:63]: would it make more sense to NOT have a setter at all? Or is this something that JDBI doesnt like

Comment (mosabua) [reply gateway-ha/src/test/java/io/trino/gateway/ha/router/TestResourceGroupsManager.java:63]: Or do we want another ID that we generate (like a UUID string or so?)

**Review by mosabua — APPROVED** [review]
Looks good. Just have some questions and minor updates to comments.

File: gateway-ha/src/test/java/io/trino/gateway/ha/router/TestResourceGroupsManager.java (hunk 2 above)
Comment (oneonestar) [reply gateway-ha/src/test/java/io/trino/gateway/ha/router/TestResourceGroupsManager.java]: I agree that's confusing. I'll delete these two lines.
This setter will remain for Jackson deserialization.

**Review by mosabua — APPROVED** [review]
Looks good. Merging this.

**Pull request merged**